processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -108,4 +110,22 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  //simple merge strategy on query granularity that checks if all are equal or else
+  //returns null. this can be improved in future but is good enough for most use-cases.
+  public static QueryGranularity mergeQueryGranularities(List<QueryGranularity> toMerge)
```

Comment:
i think you meant it for segmentMetadata query result and not to store that way in segment metdata. however, while doing merging for segmentMetadata query, if there were 2 segments with different query granularity and overlapping interval, then they would become null as well.